docs: Create CONTRIBUTING.md (#125)
ceph/ceph-nvmeof: CONTRIBUTING.md

# Welcome to Ceph NVMe-oF Gateway contributing guide <!-- omit in toc -->

## New contributor guide

The Ceph NVMe-oF Gateway project pivots around 2 other major Open Source projects:
- [SPDK](https://spdk.io/) [[source code](https://github.com/spdk/spdk/)], whih internally relies on the DPDK project.

> - [Ceph](https://ceph.com/) [[source code](https://github.com/ceph/ceph)].

Reviewer: ```suggestion
- [Ceph](https://ceph.io/) [[source code](https://github.com/ceph/ceph)].
```

Author: Thanks, @idryomov! I'll apply your suggestions!

## Reporting issues

> If you found an issue, please identify first whether the issue comes from the NVMe-oF Gateway, or any of the underlying components:

Reviewer 2: s/found/find/

s/please identify first/identify/

Author: Thanks @zdover23! So from what I understand here, I should avoid 'please' in general, right? I'll remove those.

* For NVMe-oF Gateway issues (usually a Python traceback), please check if [the issue has already been reported](https://github.com/ceph/ceph-nvmeof/issues).
  If not, please [open a new one](https://github.com/ceph/ceph-nvmeof/issues/new).
* For SPDK-related issues, please [open an issue in the SPDK Github Issues](https://github.com/spdk/spdk/issues/).

> * For Ceph-related issues, please [open an issue in the Ceph issue tracker](https://tracker.ceph.com/).

Reviewer: ```suggestion
* For Ceph-related issues, please [open an issue in the Ceph Tracker](https://tracker.ceph.com/).
```

## Submitting changes

### Coding conventions

This project follows:
* Python coding guidelines ([PEP-8](https://peps.python.org/pep-0008/)).
* [gRPC and Protocol Buffers](https://grpc.io/docs/what-is-grpc/introduction/).

### Commit format

When committing your changes, please:
* Sign-off (`git commit -s`), which will automatically add the following trailer to your commit: `Signed-off-by: FirstName LastName <email@example.com>`.
  This constitues your [Developer Certificate of Origin](https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin), and is enforced by a [CI check](https://probot.github.io/apps/dco/). 

> * Follow the [Conventional Commit syntax](https://www.conventionalcommits.org/en/v1.0.0/). This is not yet enforce via CI checks.

Reviewer: ```suggestion
* Follow the [Conventional Commit syntax](https://www.conventionalcommits.org/en/v1.0.0/). This is not yet enforced via CI checks.
```

### Testing

TODO.

### Pull request

Refer to Ceph's ["Submitting Patches to Ceph"](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst) documentation, with the difference that:
* This project uses Github Issues instead of the Ceph Tracker,

> * Therefore commits and pull request should use the `Fixes #12345` syntax (where `#12345` is the Gituub issue number), instead of the `Fixes: https://tracker.ceph.com...`.

Reviewer: ```suggestion
* Therefore commits and pull requests should use the `Fixes: #12345` syntax (where `#12345` is the Github issue number), instead of the `Fixes: https://tracker.ceph.com/...`.
```

# Welcome to Ceph NVMe-oF Gateway contributing guide <!-- omit in toc -->

## New contributor guide

The Ceph NVMe-oF Gateway project pivots around 2 other major Open Source projects:

> - [SPDK](https://spdk.io/) [[source code](https://github.com/spdk/spdk/)], whih internally relies on the DPDK project.

Reviewer: ```suggestion
- [SPDK](https://spdk.io/) [[source code](https://github.com/spdk/spdk)], which internally relies on the DPDK project.
```

Author: omg, I'll add a spellchecking action to avoid this! Thanks!

- [Ceph](https://ceph.io/) [[source code](https://github.com/ceph/ceph)].

## Engage with the community

Besides the [regular Ceph community channels](https://ceph.io/en/community/connect/), the NVMe-oF
Gateway team can be specifically reached at:
- [Ceph Slack workspace](https://ceph-storage.slack.com/): `#nvmeof` channel,
- [Weekly Sync

>   Meeting](https://calendar.google.com/calendar/event?action=TEMPLATE&tmeid=bzZpcWFvaG1pNnVkczZuYWdncmViZjgxaG5fMjAyMzA1MDlUMTQwMDAwWiA5dHM5YzdsdDd1MXZpYzJpanZ2cXFsZnBvMEBn&tmsrc=9ts9c7lt7u1vic2ijvvqqlfpo0%40group.calendar.google.com&scp=ALL): every Tuesday at 10:00am-10:30am (ET/EST). Past minutes and topic proposals can be found [here](https://pad.ceph.com/p/rbd_nvmeof).

Reviewer: It feels like this link along with the spelled out time can easily go stale.  I would just refer to the pad for the schedule and the meeting details.

Author: I wanted to make adding the meeting to their calendars a one-click thing for contributors, since handling meetings and timezones is usually a pain. But, anyway, I'll remove that and rely on the etherpad.
